### PR TITLE
Add markdownlint cli for local markdown linting

### PR DESCRIPTION
**Pull request by stevenguh**
This PR adds local lint command. Repurposing the deploy action to run the `npm lint` instead a dedicate gh action for lint so local env and gh env share the same linting code and process as well re-using the `npm install` gh action process

### CODE REVIEW

**Comment by stevenguh** [comment]
However, this might not work for the other two branches

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "build": "docusaurus build",
     "swizzle": "docusaurus swizzle",
     "deploy": "docusaurus deploy",
-    "serve": "docusaurus serve"
+    "serve": "docusaurus serve",
+    "lint": "markdownlint **/*.md --config .markdownlint.yaml --ignore node_modules --ignore docs/default-keybindings.md"
```
Comment (MarcoIeni) [inline package.json:12]:
```suggestion
    "lint": "markdownlint '**/*.md' --config .markdownlint.yaml --ignore node_modules --ignore docs/default-keybindings.md"
```

**Comment by marcoieni** [comment]
> However, this might not work for the other two branches

Maybe it is enough to put it in the ci.yml file like it was before with the rules specified on top of it.

Anyway it's not working. The current README contains errors that are not detected:
image

**Comment by stevenguh** [comment]
Finally, took many tries to get the gh action working the way I wanted to. The current actions will only require one action to run all the necessary steps:
- All pushes, pull requests will run the build and lint steps
- Only push to master will run the deploy steps in addition to build and lint
- Lint is a local runnable command directly with npm script (`npm run lint`)
- Only one `npm install` for action so ideally save some gh-action hours?

> Anyway it's not working. The current README contains errors that are not detected:

Good catch!! This apparently was due to a pair of missing quotes in the command (`'**/*.md'`)

**Comment by stevenguh** [comment]
This gh action business is tricky. To avoid duplicate action I can't use `on: [push, pull_request]`. I updated so the ci action so it will only run on push/pr to `master`, `which-key-next`, `vspacecode-next`. This way it should prevent action being run twice if someone push to a branch of a pr to one of those branches (e.g. this pr).

However, duplication can still happen. For example, if there is open pr from `which-key-next` to `master`, and someone push to `which-key-next`.

We might able to avoid duplication completely with https://github.com/fkirc/skip-duplicate-actions/#skip-duplicate-workflow-runs. However, I think the current setup is good enough (and simple) that we shouldn't see duplication that often.

File: README.md [hunk 2]
```diff
@@ -23,3 +23,11 @@ npm run build
...
 
 This command generates static content into the `build` directory and can be served using any static contents hosting service.
+
+## Lint
+
+```sh
+npm run lint
+```
+
+This command runs markdown lint
```
Comment (marcoieni) [inline README.md]:
```suggestion
This command runs [markdown lint](https://github.com/igorshubovych/markdownlint-cli). Use it to check if the markdown you have written adheres to [our](.markdownlint.yaml) style.
```
I hope the relative link will work 😅

**Comment by marcoieni** [comment]
Great job! Apart for that small comment I left it looks good to me, so you can merge it after you accepted/rejected that suggestion.

**Pull request merged**